src/parser/core/modules/EntityStatuses.ts

```diff
@@ -0,0 +1,142 @@
+			statusRanges.push({timestamp: statusEvent.end ?? this.parser.currentTimestamp, type: REMOVE, stackHistory: statusEvent.stackHistory})
+			return statusRanges
+		}, [])
+	}
+
+	splitStatusEventForInvulns(statusEvent: BuffTrackingEvent) {
+		const eventToAdjust = {...statusEvent}
+		const adjustedEvents = [eventToAdjust]
+		if (!statusEvent.end) {
+			this.debug('Unfinished status event detected.  Applying ability duration after last refresh event.')
+			const statusInfo = getDataBy(STATUSES, 'id', statusEvent.ability.guid)
+			if (statusInfo?.duration) {
+				statusEvent.end = statusEvent.lastRefreshed + statusInfo.duration * 1000
+				this.debug(`Updating status event for status ${statusInfo.name}.  Adding ${statusInfo.duration} seconds, effective end time set to ${this.parser.formatTimestamp(statusEvent.end, 1)}`)
+			} else {
+				this.debug(`No matching status duration information found for status ${statusEvent.ability.guid}, setting to end of fight so invuln detection can clip the end to when the target went untargetable`)
+				statusEvent.end = this.parser.fight.end_time
+			}
+		}
+		const target = String(statusEvent.targetID)
+		const invulns = this.invuln.getInvulns(target, statusEvent.start, statusEvent.end, 'invulnerable')
+
+		invulns.forEach((invuln: TODO) => {
+			this.debug(`Target was detected as invulnerable during duration of status.  Invulnerable from ${this.parser.formatTimestamp(invuln.start, 1)} to ${this.parser.formatTimestamp(invuln.end, 1)}`)
+			if (invuln.start < statusEvent.start && invuln.end >= statusEvent.start) {
+				this.debug('Invuln clipped the start of the range - changing beginning of event')
+				eventToAdjust.start = invuln.end
+			} else {
+				this.debug('Invuln clipped the end of the range or split the range - changing end of event')
+				eventToAdjust.end = invuln.start
+				eventToAdjust.stackHistory.splice(-1, 1, {stacks: 0, timestamp: invuln.start, invuln: true})
+
+				if (invuln.end < eventToAdjust.end!) {
+					this.debug('Invuln split the range - synthesizing second event for status time after invuln')
+					// Invuln ended before the status ended - create a second status for the time after the invuln ended
+					const newStackHistory = statusEvent.stackHistory.slice(0, -1)
+					const stacksBeforeInvuln = newStackHistory[newStackHistory.length - 1].stacks
+					newStackHistory.splice(-1, 1,
+						{stacks: 0, timestamp: invuln.start, invuln: true},
+						{stacks: stacksBeforeInvuln, timestamp: invuln.end, invuln: false},
+						{stacks: 0, timestamp: statusEvent.end!, invuln: false})
+
+					adjustedEvents.push({
+						...statusEvent,
+						start: invuln.end,
+						lastRefreshed: invuln.end,
+						end: statusEvent.end,
+						stackHistory: newStackHistory,
+					})
+				}
+			}
+		})
+
+		adjustedEvents.forEach(event => this.debug(`Active time after adjusting for invulns - start at: ${this.parser.formatTimestamp(event.start)} - end at: ${this.parser.formatTimestamp(event.end!)}`))
```

Comment:
mmh - this is fine, but i should probably add an overload signature to `debug` for a function arg so that potentially expensive argument computations can be done in debug only.